lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -2800,7 +2800,9 @@ FuncInfo* PostVisitFunction(ParseNode* pnode, ByteCodeGenerator* byteCodeGenerat
 
             parentFunc->SetChildHasWith();
 
-            if (parentFunc->GetBodyScope()->GetHasOwnLocalInClosure())
+            if (parentFunc->GetBodyScope()->GetHasOwnLocalInClosure() ||
+                (parentFunc->GetParamScope()->GetHasOwnLocalInClosure() &&
```

Comment:
>parentFunc->GetParamScope()->GetHasOwnLocalInClosure() [](start = 17, length = 54)

If the param scope has own local in closure then shouldn't it always be split? Else shouldn't we clear the HasOwnLocalInClosure flag when we merge both scopes?